pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -498,6 +508,8 @@ public void run() {
             ServerGauge.LAST_REALTIME_SEGMENT_COMPLETION_DURATION_SECONDS,
             TimeUnit.MILLISECONDS.toSeconds(now() - initialConsumptionEnd));
       }
+      // Set the highest offset consumed to 0, the next consumer will restart it.
+      _serverMetrics.setValueOfTableGauge(_metricKeyName, ServerGauge.HIGHEST_KAFKA_OFFSET_CONSUMED, 0L);
```

Comment:
For the duration of the state transition, this metric will be 0, which is not correct.  I think you are repurposing HIGHEST_OFFSET_CONSUMED for other purposes. Perhaps a new metric will be more appropriate.